lib/queue.rb

```diff
@@ -1,31 +1,54 @@
 class Queue
-
+  
   def initialize
     # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+      
+    end
+    
+    if @front == @back
+      raise ArgumentError, "Queue is full."
+    end
+    
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
+    if front.nil?
+      @front = (@front + 1) % @store.length
+    end
+    
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if (size == 0)
+      return nil
+    end
+    
+    temp = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+    return temp
   end
-
+  
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
-
+  
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.compact.size
   end
-
+  
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return size == 0
   end
-
+  
   def to_s
-    return @store.to_s
+    return @store[@front...@back].to_s
```

Comment:
this only works if front < back

Something like this is better.

```suggestion
    return @store[@front...@back].to_s if @front < @back
    return [@front...@store.length] + @store[0...@back]
```